### PR TITLE
Cyborg Language Module

### PR DESCRIPTION
**Pull request by MisterLayne**
Adds a cyborg language module allowing cyborg to speak full-range of languages. (Excluding ~~LocalRoot and~~ Kiki)

### CODE REVIEW

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 1]
```diff
@@ -173,3 +173,23 @@
...
+/obj/item/borg/upgrade/language
+	name = "Language module"
+	desc = "Used to let cyborgs other than clerical or service speak a variety of languages."
+	icon_state = "cyborg_upgrade3"
+	item_state = "cyborg_upgrade"
+
+/obj/item/borg/upgrade/language/action(var/mob/living/silicon/robot/R)
+	if(..()) return 0
+
+	R.add_language(LANGUAGE_UNATHI, 1)
```
Comment (elgeonmb) [inline code/game/objects/items/robot/robot_upgrades.dm]: unathi twice?

**Comment by zekesulastin** [comment]
Isn't being able to speak other languages one of the specific features of the clerical/service borgs?

**Comment by Belsima** [comment]
+	desc = "Used to let cyborgs other than clerical or service speak a variety of languages."

**Comment by Anewbe** [comment]
## UNIT_TEST: The entry /datum/design/item/prosfab/robot_upgrade/language has invalid material type DDFAULT_WALL_MATERIAL

**Comment by Anewbe** [comment]
Wow that got bolded, weird...

**Comment by PrismaticGynoid** [comment]
And what happens if it gets removed?

**Comment by Anewbe** [comment]
I know basically nothing about Robots, is it even possible to remove upgrades?

**Comment by elgeonmb** [comment]
@Anewbe among other things module resets do it

**Comment by MisterLayne** [comment]
I don't actually think a module reset will remove this upgrade. A module reset doesn't remove any upgrades if I'm not mistaken.

**Comment by Mechoid** [comment]
Module resets do not remove upgrades to my knowledge, and I am not inclined to change this, myself. Waste of materials.

**Comment by Mechoid** [comment]
In the OP you said it doesn't add root local, though code wise it does appear to. I have no issue as Service / Clerical / AI's themselves (AI may have not been fixed on that yet) are specifically capable of doing so, so words from Elge on Trees may be needed on that. Unless you meant Root Global instead, which is in fact not included in the language upgrade.

**Comment by Anewbe** [comment]
AIs have not been fixed on that yet. I looked into it and could not get it to work, for some reason.

**Comment by Nalarac** [comment]
Module resets currently remove taser upgrades and the jetpack as those are items in the cyborg's inventory.

**Comment by MisterLayne** [comment]
Resets only remove module-related items. VTECs and the Language module will stay. The reason that I originally said that it excluded Root Local is because I just pushed it to add RootLocal. I wanted the heads-up before I added RootLocal to the speakable languages.

**Comment by Mechoid** [comment]
Ahhh, alright. Well, it looks good to me, if you've tested it / its production.

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 2]
```diff
@@ -173,3 +173,23 @@
 
 	R.emag_items = 1
 	return 1
+
+/obj/item/borg/upgrade/language
+	name = "Language module"
```
Comment (Belsima) [inline code/game/objects/items/robot/robot_upgrades.dm]: Should be lowercase.

Comment (Mechoid) [reply code/game/objects/items/robot/robot_upgrades.dm]: I'm apparently very bad at noticing capitalization. Probably should be lowercase, yeah.

**Comment by MisterLayne** [comment]
It has been tested, it's all ready to go~

**Pull request merged**